### PR TITLE
LPS-148437 Add integration tests for ObjectViewSortColumn

### PR DESCRIPTION
**Pull request by mateussgn**
Related Issue: https://issues.liferay.com/browse/LPS-148437

Related issue: https://issues.liferay.com/browse/LPS-148437

### CODE REVIEW

**Comment by liferay-forms** [comment]
Hi, you don’t need to run the CI commands anymore, I’ll do it for you and let you know if any issue happens.

**Comment by liferay-forms** [comment]
ci:test:sf

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectViewLocalServiceImpl.java [hunk 1]
```diff
@@ -284,9 +304,51 @@ private void _validate(long objectViewId, long objectDefinitionId)
...
+			_objectFieldPersistence.findByObjectDefinitionId(
+				objectView.getObjectDefinitionId());
+
+		List<String> objectFieldLabel = new ArrayList<>();
+
+		for (ObjectField objectField : objectFields) {
+			objectFieldLabel.add(objectField.getName());
+		}
+
+		List<String> sortOrders = new ArrayList<String>() {
```
Comment (marcelabc) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectViewLocalServiceImpl.java:322]: I'm afraid I have to disagree with creating this variable inside this method. Could you read the comment below?

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectViewLocalServiceImpl.java [hunk 2]
```diff
@@ -284,9 +304,51 @@ private void _validate(long objectViewId, long objectDefinitionId)
...
+				objectViewSortColumns) {
+
+			if (!objectFieldLabel.contains(
+					objectViewSortColumn.getObjectFieldName())) {
+
+				throw new ObjectViewSortColumnException(
+					"This object field does not exists");
+			}
+
+			if (!sortOrders.contains(objectViewSortColumn.getSortOrder())) {
```
[the remark this answers is not in the document]
Comment (marcelabc) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectViewLocalServiceImpl.java:339]: We can discuss it more, if you like :+1:

**Review by marcelabc — CHANGES_REQUESTED** [review]
Hey @mateussgn , how are you doing?
I finished your review! Let me know if there is any question.

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectViewLocalServiceImpl.java [hunk 3]
```diff
@@ -284,9 +304,51 @@ private void _validate(long objectViewId, long objectDefinitionId)
 		}
 	}
 
+	private void _validateSortColumn(
+			List<ObjectViewSortColumn> objectViewSortColumns,
```
Comment (marcelabc) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectViewLocalServiceImpl.java:308]: Could you sort the parameters?

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectViewLocalServiceImpl.java [hunk 4]
```diff
@@ -284,9 +304,51 @@ private void _validate(long objectViewId, long objectDefinitionId)
...
+			ObjectView objectView)
+		throws ObjectViewSortColumnException {
+
+		List<ObjectField> objectFields =
+			_objectFieldPersistence.findByObjectDefinitionId(
```
Comment (marcelabc) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectViewLocalServiceImpl.java:313]: Could you use the method from the LocalService, please?

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectViewLocalServiceImpl.java [hunk 5]
```diff
@@ -284,9 +304,51 @@ private void _validate(long objectViewId, long objectDefinitionId)
...
+				objectView.getObjectDefinitionId());
+
+		List<String> objectFieldLabel = new ArrayList<>();
```
Comment (marcelabc) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectViewLocalServiceImpl.java:316]: When the variable is from List type, the variable name shouldn't be in plural?

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectViewLocalServiceImpl.java [hunk 6]
```diff
@@ -284,9 +304,51 @@ private void _validate(long objectViewId, long objectDefinitionId)
...
+		};
+
+		for (ObjectViewSortColumn objectViewSortColumn :
+				objectViewSortColumns) {
+
+			if (!objectFieldLabel.contains(
+					objectViewSortColumn.getObjectFieldName())) {
+
+				throw new ObjectViewSortColumnException(
+					"This object field does not exists");
```
[the remark this answers is not in the document]
Comment (mateussgn) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectViewLocalServiceImpl.java:336]: Double-checked. The pattern "There is no..." is more often used, so I changed to it.

**Comment by mateussgn** [comment]
ci:test:relevant

**Comment by mateussgn** [comment]
ci:stop

**Comment by mateussgn** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
Hey @mateussgn, can you please check if these failures are related to your changes? Thanks

**Comment by mateussgn** [comment]
ci:test:forward

**Comment by mateussgn** [comment]
ci:forward

**Comment by liferay-forms** [comment]
Hey @mateussgn, can you please fix the source formatter issues and resend? Thanks

**Pull request closed without merging**